src/pdf/ShoppingList/shoppingListPdfDataProps.ts

```diff
@@ -40,8 +40,8 @@ const getQuantityAndNotes = (
     if (size >= 10) {
         size = 10;
     }
-    const sizeQuantity = `${size}_quantity` as keyof Schema["lists"];
-    const sizeNotes = `${size}_notes` as keyof Schema["lists"];
+    const sizeQuantity = `quantity_for_${size}` as keyof Schema["lists"];
+    const sizeNotes = `notes_for_${size}` as keyof Schema["lists"];
```

Comment:
I think these `as keyof Schema["lists"]` is what allowed this bug to happen in the first place, so let's try make this more robust.

One approach would be to define the set of allowed family sizes, and make sure the family size is one of them, before computing the quantity and notes strings. This way, we don't need to use `as` and this kind of bug won't happen again.

```ts
const allowedFamilySizes = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10] as const;
type FamilySize = (typeof allowedFamilySizes)[number];

function numberIsValidFamilySize(value: number): value is FamilySize {
    for (const allowedFamilySize of allowedFamilySizes) {
        if (allowedFamilySize === value) {
            return true;
        }
    }
    return false;
}

const getQuantityAndNotes = (
    row: Schema["lists"],
    size: number
): Pick<Item, "quantity" | "notes"> => {
    if (size >= 10) {
        size = 10;
    }

    if (!numberIsValidFamilySize(size)) {
        // return an error as we normally do
    }

    const sizeQuantity: keyof Schema["lists"] = `quantity_for_${size}`;
    const sizeNotes: keyof Schema["lists"] = `notes_for_${size}`;
    return {
        quantity: row[sizeQuantity]?.toString() ?? "",
        notes: row[sizeNotes]?.toString() ?? "",
    };
};
```